apps/web/src/pages/integrations/UpdateProviderPage.tsx

```diff
@@ -1,42 +1,355 @@
+
+  if (isLoading || areEnvironmentsLoading) {
+    return (
+      <SideBarWrapper dark={isDark}>
+        <Stack
+          align="center"
+          justify="center"
+          sx={{
+            height: '100%',
+          }}
+        >
+          <Loader color={colors.error} size={32} />
+        </Stack>
+      </SideBarWrapper>
+    );
+  }
 
   if (selectedProvider === null) {
     return null;
   }
 
-  return <SideBarWrapper dark={isDark}></SideBarWrapper>;
+  return (
+    <SideBarWrapper dark={isDark}>
+      <Form
+        name={'connect-integration-form'}
+        noValidate
+        onSubmit={(e) => {
+          handleSubmit((values) => {
+            refetch();
+          });
+          e.preventDefault();
+        }}
+      >
+        <Group spacing={5}>
+          <ProviderImage providerId={selectedProvider?.providerId} />
+          <Controller
+            control={control}
+            name="name"
+            defaultValue=""
+            render={({ field }) => {
+              return (
+                <NameInput
+                  {...field}
+                  value={field.value !== undefined ? field.value : selectedProvider?.displayName}
+                  data-test-id="provider-instance-name"
+                  placeholder="Enter instance name"
+                />
+              );
+            }}
+          />
+          <ActionIcon
+            variant={'transparent'}
+            onClick={() => {
+              navigate('/integrations');
+            }}
+          >
+            <Close color={colors.B40} />
+          </ActionIcon>
+        </Group>
+        <Group mb={16} mt={16} spacing={16}>
+          <IntegrationChannel
+            name={CHANNEL_TYPE_TO_STRING[selectedProvider?.channel || ChannelTypeEnum.EMAIL]}
+            type={selectedProvider?.channel || ChannelTypeEnum.EMAIL}
+          />
+          <IntegrationEnvironmentPill
+            name={
+              environments?.find((environment) => environment._id === selectedProvider?.environmentId)?.name ||
+              'Development'
+            }
+          />
+        </Group>
+        <CenterDiv>
+          <When truthy={!haveAllCredentials}>
+            <WarningMessage spacing={12}>
+              <DisconnectGradient />
+              <div>
+                Set up credentials to start sending notifications
+                <br />
+                <a href={selectedProvider?.docReference} target="_blank" rel="noopener noreferrer">
+                  Explore set-up guide
+                </a>
+              </div>
+            </WarningMessage>
+          </When>
+          <ActiveWrapper active={isActive}>
+            <Controller
+              control={control}
+              name="active"
+              render={({ field }) => (
+                <Switch
+                  checked={isActive}
+                  label={isActive ? 'Active' : 'Disabled'}
+                  data-test-id="is_active_id"
+                  {...field}
+                />
```

Comment:
```suggestion
                <Switch
                  checked={field.value}
                  label={field.value ? 'Active' : 'Disabled'}
                  data-test-id="is_active_id"
                  {...field}
                />
```
And then we can remove the watch over active